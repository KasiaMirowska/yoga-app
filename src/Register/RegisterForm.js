import React from 'react';
import AuthCalls from '../services/Auth_service';
import ValidationError from './ValidationError';


export default class RegisterForm extends React.Component {
    state = {
        error: null,
        full_name: {
            value: '',
        },
        user_name: {
            value: '',
            touched: false
        },
        password: {
            value: '',
            touched: false
        },
    }

    handleFullName = (name) => {
        console.log(name)
        this.setState({ full_name: { value: name, touched: true } })
    }

    handleUserName = (name) => {
        console.log(name)
        this.setState({ user_name: { value: name, touched: true } })
    }

    handlePassword = (password) => {
        console.log(password)
        this.setState({ password: { value: password, touched: true } })
    }

    validateUserName = () => {
        const name = this.state.user_name.value.trim;
        if (name.length === 0) {
            return 'User name is required';
        } else if (name.length < 3) {
            return 'User name must be at least 3 characters long';
        }
    }

    validatePassword = () => {
        const password = this.state.password.value.trim();
        if (password.length === 0) {
            return 'Password is required';
        } else if (password.length < 6 || password.length > 72) {
            return 'Password must be between 6 and 72 characters long';
        } else if (!password.match(/[0-9]/)) {
            return 'Password must contain at least one number';
        }
    }
    handleSubmit = (e) => {
        e.preventDefault();
        const { full_name, user_name, password } = this.state;
        const newUser = {
            fullname: full_name.value,
            username: user_name.value,
            password: password.value,
        }
        console.log(newUser)
        AuthCalls.postUser(newUser)
        .then(user => {
            full_name.value ='';
            user_name.value = '';
            password.value = '';
            this.props.onRegistrationSuccess()
        })
        .catch(res => {
            this.setState({
                error: res.error
            })
        })
    }
    render() {
        console.log(this.state.error)
        return (
            <div>
                <form onSubmit={this.handleSubmit}>
                    <label htmlFor='full_name'>
                        User Full Name:
                    </label>
                    <input name='full_name' type='text' onChange={(e) => this.handleFullName(e.target.value)} />
                    <label htmlFor='full_name'>
                        User Name:
                    </label>
                    <input name='user_name' type='text' onChange={(e) => this.handleUserName(e.target.value)} required />
                    {this.state.password.touched && <ValidationError message={this.validateUserName()} />}
                    <label htmlFor='password'>
                        Password:
                    </label>
                    <input name="password" type='text' onChange={(e) => this.handlePassword(e.target.value)} required />
                    {this.state.user_name.touched && <ValidationError message={this.validatePassword()} />}
                    <button type="reset" >
                        Cancel
                    </button>
                    <button
                        type='submit'
                        // disabled = {
                        //     this.validateUserName() || this.validatePassword()
                        // }
                        >
                        Submit
                            </button>
                </form>
            </div>
        )
    }
}