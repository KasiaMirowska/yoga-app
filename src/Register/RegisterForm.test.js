import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import RegisterForm from './RegisterForm';
import {BrowserRouter as BR} from 'react-router-dom';

describe('RegisterForm component', () => {
    const props = {match: {params: {path: '/register'}}}
    
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><RegisterForm {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
    it('renders UI as expected', () => {
        const register = renderer.create(<BR><RegisterForm /></BR>);
        expect(register.toJSON()).toMatchSnapshot();
    })

})