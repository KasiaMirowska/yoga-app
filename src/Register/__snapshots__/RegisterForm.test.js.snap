// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm component renders UI as expected 1`] = `
<div
  className="login"
>
  <div
    className="error"
  />
  <form
    onSubmit={[Function]}
  >
    <div
      className="section"
    >
      <input
        name="full_name"
        onClick={[Function]}
        placeholder="full name"
        type="text"
      />
    </div>
    <div
      className="section"
    >
      <input
        name="user_name"
        onClick={[Function]}
        placeholder="username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="section"
    >
      <input
        name="password"
        onClick={[Function]}
        placeholder="password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="login-button"
      onClick={[Function]}
      type="reset"
    >
      Cancel
    </button>
    <button
      className="login-button"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
